pkg/controller/elasticsearch/nodespec/readiness_probe.go

```diff
@@ -58,3 +57,23 @@ else
 	exit 1
 fi
 `
+
+const PreStopHookScriptConfigKey = "pre-stop-hook-script.sh"
+const PreStopHookScript = `#!/usr/bin/env bash
+
+# This script will wait for up to $MAX_WAIT_SECONDS for $POD_IP to disappear from DNS record,
+# then it will wait additional $ADDN_WAIT_SECONDS and exit. This slows down the process shutdown
+# and allows to make changes to the pool gracefully, without blackholing traffic when DNS
+# contains IP that is already inactive. Assumes $SERVICE_NAME and $POD_IP env variables are defined.
+
+MAX_WAIT_SECONDS=20 # max time to wait for pods IP to disappear from DNS
```

Comment:
IIUC `MAX_WAIT_SECONDS` should not be greater than the grace period _(30 sec by default)_ ? And it should give some time for the container to gracefully handle the SIGTERM signal ?

Reply:
👍 maybe that a quick note about the interaction with the grace period would be nice